runtime/drivers/clickhouse/olap.go

```diff
@@ -209,7 +209,7 @@ func (c *connection) AlterTableColumn(ctx context.Context, tableName, columnName
 }
 
 // CreateTableAsSelect implements drivers.OLAPStore.
-func (c *connection) CreateTableAsSelect(ctx context.Context, name string, view bool, sql string, tableOpts map[string]any) error {
+func (c *connection) CreateTableAsSelect(ctx context.Context, name string, view bool, sql, beforeCreateSQL, afterCreateSQL string, tableOpts map[string]any) error {
```

Comment:
It seems like it's not executing the before/after SQL here?

Reply:
Yeah clickhouse is a planned follow up. Need to evaluate if these queries should this be run by the same connection. If yes, then some refactors are required. Currently there are multiple queries being executed and those can be executed on any connection. 
Also added a draft issue in my bucket to track this.